src/Sulu/Bundle/ContentBundle/Resources/public/js/components/content/settings/main.js

```diff
@@ -360,12 +372,66 @@ define(['app-config', 'sulusecurity/components/users/models/user'], function(App
                 data.shadowBaseLanguage = baseLanguages[0];
             }
 
+            if (!this.validate(data)) {
+                this.sandbox.emit('sulu.labels.warning.show', 'form.validation-warning', 'labels.warning');
+
+                return;
+            }
+
             this.data = this.sandbox.util.extend(true, {}, this.data, data);
 
             // nav contexts not extend
             this.data.navContexts = data.navContexts;
 
             this.sandbox.emit('sulu.content.contents.save', this.data);
+        },
+
+        validate: function(data) {
+            // remove validation classes
+            this.sandbox.dom.removeClass(constants.internalLink.titleContainer, constants.validateErrorClass);
+            this.sandbox.dom.removeClass(constants.internalLink.linkContainer, constants.validateErrorClass);
+            this.sandbox.dom.removeClass(constants.externalLink.titleContainer, constants.validateErrorClass);
+            this.sandbox.dom.removeClass(constants.externalLink.linkContainer, constants.validateErrorClass);
+
+            if (data.nodeType === TYPE_INTERNAL) {
+                return this.validateInternal(data);
+            } else if (data.nodeType === TYPE_EXTERNAL) {
+                return this.validateExternal(data);
+            }
+
+            return true;
+        },
+
+        validateInternal: function(data) {
+            var result = true;
```

Comment:
Minor code style issue: I would not introduce this variable:

```javascript
            if (!data.title) {
                this.sandbox.dom.addClass(constants.internalLink.titleContainer, constants.validateErrorClass);
                return false;
            }

            if (!data.internal_link) {
                this.sandbox.dom.addClass(constants.internalLink.linkContainer, constants.validateErrorClass);
                return false;
            }

            return true;
```

Should do the same job, with less lines, memory and in less time :wink: